pcm-core.cpp

```diff
@@ -290,16 +291,17 @@ int main(int argc, char * argv[])
 	int calibrated = PCM_CALIBRATION_INTERVAL - 2; // keeps track is the clock calibration needed
 	unsigned int numberOfIterations = 0; // number of iterations
 	string program = string(argv[0]);
-	EventSelectRegister regs[4];
+	EventSelectRegister *regs;
 	PCM::ExtendedCustomCoreEventDescription conf;
 	bool show_partial_core_output = false;
 	std::bitset<MAX_CORES> ycores;
 
+        PCM * m = PCM::getInstance();
+
+        regs = (EventSelectRegister*)malloc(sizeof(EventSelectRegister) * m->getMaxCustomCoreEvents());
 	// Occasionally the memory is not properly nulled because counters appear to
 	// be programmed even without given arguments so making really sure
-	memset( &regs, 0, sizeof(regs) );
-
-    PCM * m = PCM::getInstance();
+        memset(regs, 0, sizeof(EventSelectRegister) * m->getMaxCustomCoreEvents());
```

Comment:
the original code used memset as well which I actually don't like (in C++ there are better less error-prone techniques to initialize objects rather than using malloc/memset). Could you please add "EventSelectRegister() : value(0) {}" default constructor into struct EventSelectRegister in types.h and drop this memset? This should do the job

Reply:
Dropped the memset and added the default constructor.